regex/src/main/scala/regex/Regex.scala

```diff
@@ -15,14 +29,53 @@ package regex
   */
 
 /** Simplifies a regular language */
-def simplify(lang: RegularLanguage): RegularLanguage = ???
+def simplify(lang: RegularLanguage): RegularLanguage = 
+  lang match {
+    case Concat(Epsilon, input) => simplify(input)
+    case Concat(input, Epsilon) => simplify(input)
+    case Concat(Empty, input) => Empty
```

Comment:
I put in wildcards instead of inputs for these Concat cases since they returned Empty for all inputs.